File: java/268 Missing Number/MissingNumber.java

/**
 * Given an array containing n distinct numbers taken from 0, 1, 2, ..., n, find the one that is missing from the array.
 * For example, Given nums = [0, 1, 3] return 2.
 */
public class MissingNumber {
    public int missingNumber(int[] nums) {
        int sum = 0, n = nums.length;
        for (int i : nums)
            sum += i;
        return (n * (n + 1)) / 2 - sum;
    }
}

File: swift/268 Missing Number/MissingNumber.swift

/**
 * Given an array containing n distinct numbers taken from 0, 1, 2, ..., n, find the one that is missing from the array.
 * For example, Given nums = [0, 1, 3] return 2.
 */
class Solution {
    func missingNumber(_ nums: [Int]) -> Int {
        var miss = 0, i = 0
        for var num in nums {
            i += 1
            miss += i - num
        }
        return miss
    }
}
